package/kedro_viz/integrations/kedro/sqlite_store.py

```diff
@@ -42,10 +43,22 @@ def to_json(self) -> str:
         """Returns session_store information in json format after converting PosixPath to string"""
         session_dict = {}
         for key, value in self.data.items():
-            if _is_json_serializable(value):
+            # Get branch info through git_sha
+            if key == "git":
+                sha = [v for k, v in value.items() if k == "commit_sha"]
+                git = subprocess.run(
```

Comment:
Kedro comes out of the box with gitpython installed: https://github.com/quantumblacklabs/kedro/blob/master/requirements.txt#L7 -- can you check to see if they any API we can leverage instead of doing subprocess?

Reply:
Thanks, i also realised that if we do it using git-python. We don't really need 'sha' lookup. Because branch name gets added during kedro-run to the session_store.db and if i do repo.active_branch, I get the branch name they were working from.

Reply:
@limdauto  - let me know if this works and I can update the tests

Reply:
Also, should we write branch_name to the repo when doing kedro run. or should we retrieve branch_name from sha during kedro-viz.
Initialy I was doing it on development environment so ofcourse if i tried to run github commands -- it would run on kedro-viz github rather than kedro-project github. But for production mode it would be fine because both would be run from kedro project github

Reply:
@limdauto 

Butting in here because Rashida asked me about something else on this PR and then I just spotted this comment...

Even though gitpython is a requirement of kedro, this doesn't mean that the git executable exists. If it doesn't then `import git` will crash. And if the git executable _does_ exist but we're not in a git repo, this code should be written so it doesn't crash.

So unless git is a hard requirement of kedro-viz (which I don't think it should be), we should probably:
* move the `import git` inline
* wrap the whole thing in some error catching code

See https://github.com/quantumblacklabs/private-kedro/pull/1284 🙂

Reply:
Worth checking also what happens if `import git` works but you're not in a repo - what does `git.Repo` do then? Again this should return empty data rather than crash.

Reply:
I can confirm it now works when you're not in any repo. It just doesn't have git in the session data